test/sanity/validate-modules/main.py

```diff
@@ -1001,8 +887,167 @@ def _validate_docs(self):
                 )
 
             if metadata:
-                self._validate_docs_schema(metadata, metadata_1_1_schema(deprecated),
+                self._validate_docs_schema(metadata, metadata_1_1_schema(),
                                            'ANSIBLE_METADATA', 316)
+                # We could validate these via the schema if we knew what the values are ahead of
+                # time.  We can figure that out for deprecated but we can't for removed.  Only the
+                # metadata has that information.
+                if 'removed' in metadata['status']:
+                    removed = True
+                if 'deprecated' in metadata['status']:
+                    deprecated = True
+                if (deprecated or removed) and len(metadata['status']) > 1:
+                    self.reporter.error(
+                        path=self.object_path,
+                        code=333,
+                        msg='ANSIBLE_METADATA with status "deprecated" or "removed" cannot have any other status'
```

Comment:
Suggestion to make it clearer that `status` is an attribute
`ANSIBLE_METADATA.status must be exactly one of "deprecated" or "removed"'`